File: C#/T1/Project/BackAlleyDiceShooter/Player.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace BackAlleyDiceShooter
{
    class Player
    {
        // Player members
        private static int startBalance = 200;

        private static string playerName;

        private static int playerBalance;

        private static int playerBetAmount;

        private static int playerChoice;

        // Player Constructor
        public Player(string pName)
        {
            playerName = pName;
            playerBalance = startBalance;
        }

        // Get Functions
        public static string getName()
        {
            return playerName;
        }

        public static int getBalance()
        {
            return playerBalance;
        }

        public static int getBetAmount()
        {
            return playerBetAmount;
        }

        public static int getChoice()
        {
            return playerChoice;
        }

        // Set Functions
        public static void setMoney(int amount)
        {
            playerBalance += amount;
        }

        public static void setBetAmount(int amount)
        {
            playerBetAmount = amount;
        }

        public static void setChoice(int choice)
        {
            playerChoice = choice;
        }

        // Reset Balance
        public static void resetMoney()
        {
            playerBalance = startBalance;
        }
    }
}

File: C++/T1/BackAlleyDiceShooter/BackAlleyDiceShooter/Player.cpp
#include "Player.h"

#define START_BALANCE 200

Player::Player()
	: playerName(""),
	playerBalance(START_BALANCE),
	playerBetAmount(0),
	playerChoice(0)
{}

Player::~Player()
{
}

const std::string Player::getName() const
{
	return playerName;
}

const int Player::getBalance() const
{
	return playerBalance;
}

const int Player::getBetAmount() const
{
	return playerBetAmount;
}

const int Player::getChoice() const
{
	return playerChoice;
}

void Player::setName(std::string pName)
{
	playerName = pName;
}

void Player::setMoney(int amount)
{
	playerBalance += amount;
}

void Player::setBetAmount(int amount)
{
	playerBetAmount = amount;
}

void Player::setChoice(int choice)
{
	playerChoice = choice;
}

void Player::resetMoney()
{
	playerBalance = START_BALANCE;
}

